### PR TITLE
Clarify path prefix rule

### LINKED ISSUE #732: Path prefix?
The explainer [states](https://github.com/WICG/webpackage/blob/fcb50a7ed74c521f94adaef1955831beba302d19/explainers/subresource-loading.md#script-based-api_:

> Records the resources and delays fetching a subresource specified there if a subresource's origin is the [same origin](https://html.spec.whatwg.org/#same-origin) as the bundle's origin and its [path](https://url.spec.whatwg.org/#concept-url-path) contains the bundle's path as a prefix.

However, none of the examples have the bundle's path contained as a prefix of the resource path. E.g. `dir/subresources.wbn` is not a prefix of `dir/a.js`.

Which is correct, the examples or the explainer text?

### PR DESCRIPTION
**Pull request by hayatoito**
Fix #732.

***
<a href="https://pr-preview.s3.amazonaws.com/hayatoito/webpackage/pull/740.html" title="Last updated on Apr 28, 2022, 7:35 AM UTC (0ec5acb)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/webpackage/740/38390d9...hayatoito:0ec5acb.html" title="Last updated on Apr 28, 2022, 7:35 AM UTC (0ec5acb)">Diff</a>

### CODE REVIEW

File: subresource-loading.bs [hunk 1]
```diff
@@ -605,18 +605,16 @@ To <dfn>find a matching web bundle registration</dfn> given [=request=]
 
    1. Let |rule| be |registration|'s [=web bundle registration/rule=].
 
-   2. If |url| doesn't meet a path restriction rule, given |registration|'s
-      [=web bundle registration/fetch entry=]'s [=web bundle fetch
-      entry/source=], then return null.
+   2. Let |allowed path| be the result of [shortening]
+      (https://url.spec.whatwg.org/#shorten-a-urls-path) |registration|'s [=web
```
Comment (irori) [inline subresource-loading.bs:609]: This is not linkified in the rendered HTML...

File: subresource-loading.bs [hunk 2]
```diff
@@ -605,18 +605,16 @@ To <dfn>find a matching web bundle registration</dfn> given [=request=]
...
+      bundle registration/fetch entry=]'s [=web bundle fetch entry/source=].
 
-      Issue: Clarify path restriction rule in algorithm.
+   3. If |url| doesn't start with |allowed path|, then return null.
```
Comment (irori) [inline subresource-loading.bs:612]: What about `uuid-in-package:` URLs?
In Chromium implementation, we skip step 2 and step 3 if |url|'s scheme is `uuid-in-package`.

File: subresource-loading.bs (hunk 1 above)
Comment (hayatoito) [reply subresource-loading.bs]: Sorry. Fixed.

File: subresource-loading.bs (hunk 2 above)
Comment (hayatoito) [reply subresource-loading.bs]: A good point. Mentioned `uuid-in-package`. Done.

**Comment by hayatoito** [comment]
Thanks for the review. Fixed. Could you review again?

File: subresource-loading.bs [hunk 3]
```diff
@@ -605,18 +608,18 @@ To <dfn>find a matching web bundle registration</dfn> given [=request=]
...
-      [=web bundle registration/fetch entry=]'s [=web bundle fetch
-      entry/source=], then return null.
+   2. If |url|'s [=url/scheme=] is not "`uuid-in-package`", then
+
+      1. Let |allowed path| be the result of [=shortening=] |registration|'s
+         [=web bundle registration/fetch entry=]'s [=web bundle fetch
+         entry/source=].
 
-      Issue: Clarify path restriction rule in algorithm.
+      2. If |url| doesn't start with |allowed path|, then return null.
```
Comment (irori) [inline subresource-loading.bs:617]: Shouldn't we continue the registration loop instead of returning null?

Comment (hayatoito) [reply subresource-loading.bs]: Ops! Fixed!

**Review by irori — APPROVED** [review]
LGTM

**Pull request merged**